lib/taskana-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java

```diff
@@ -0,0 +1,111 @@
+
+    WorkbasketService workbasketService;
+    TaskService taskService;
+
+    @Before
+    public void before() {
+        workbasketService = taskanaEngine.getWorkbasketService();
+        taskService = taskanaEngine.getTaskService();
+    }
+
+    @After
+    public void after() throws Exception {
+        resetDb(true);
+    }
+
+    @WithAccessId(userName = "admin")
+    @Test
+    public void shouldCleanWorkbasketMarkedForDeletion() throws Exception {
+        long totalWorkbasketCount = workbasketService.createWorkbasketQuery().count();
+        assertEquals(25, totalWorkbasketCount);
+        List<WorkbasketSummary> workbaskets = workbasketService.createWorkbasketQuery()
+            .keyIn("GPK_KSC", "sort001")
+            .orderByKey(
+                BaseQuery.SortDirection.ASCENDING)
+            .list();
+        long numTasksNotCompletedInWorkbasket = getNumberTaskNotCompleted(workbaskets.get(1).getId());
+        assertEquals(numTasksNotCompletedInWorkbasket, 0);
+
+        try {
+            workbasketService.deleteWorkbasket(workbaskets.get(1).getId());
+        } catch (NotAuthorizedException | WorkbasketNotFoundException | WorkbasketInUseException | InvalidArgumentException e) {
+            // workbaskets.get(0) contains non-completed tasks and can´t be marked for deletion.
+        }
+
+        WorkbasketCleanupJob job = new WorkbasketCleanupJob(taskanaEngine, null, null);
+        job.run();
+
+        totalWorkbasketCount = workbasketService.createWorkbasketQuery().count();
+        assertEquals(24, totalWorkbasketCount);
+    }
+
+    @WithAccessId(userName = "admin")
+    @Test
+    public void shouldCleanWorkbasketMarkedForDeletionWithCompletedTasks() throws TaskanaException {
+        long totalWorkbasketCount = workbasketService.createWorkbasketQuery().count();
+        assertEquals(25, totalWorkbasketCount);
+        List<WorkbasketSummary> workbaskets = workbasketService.createWorkbasketQuery()
+            .keyIn("GPK_KSC", "sort001")
+            .orderByKey(
+                BaseQuery.SortDirection.ASCENDING)
+            .list();
+        long numTasksNotCompletedInWorkbasket = getNumberTaskNotCompleted(workbaskets.get(0).getId());
+        assertNotEquals(numTasksNotCompletedInWorkbasket, 0);
+
+        numTasksNotCompletedInWorkbasket = getNumberTaskNotCompleted(workbaskets.get(1).getId());
+        assertEquals(numTasksNotCompletedInWorkbasket, 0);
+        try {
+            workbasketService.deleteWorkbasket(workbaskets.get(1).getId());
+            workbasketService.deleteWorkbasket(workbaskets.get(0).getId());
+        } catch (NotAuthorizedException | WorkbasketNotFoundException | WorkbasketInUseException | InvalidArgumentException e) {
+            // workbaskets.get(0) contains non-completed tasks and can´t be marked for deletion.
+        }
+
+        WorkbasketCleanupJob job = new WorkbasketCleanupJob(taskanaEngine, null, null);
+        job.run();
```

Comment:
Please give it a try commenting out this line. It should still work because an empty workbasket is deleted right away.

Reply:
Did you change something regarding the acceptance tests? I don't see it actually...

Reply:
I am going to redo the tests because now they looks weird.

Reply:
I have redo the tests because they looks weird.